tests/e2e_logging.py

```diff
@@ -1588,6 +1588,61 @@ def test_committed_index(network, args):
     }
 
 
+@reqs.description("Test the commit endpoints view_history feature")
+def test_commit_view_history(network, args):
+    remote_node, _ = network.find_primary()
+    with remote_node.client() as c:
+        # the original endpoint should still work
+        res = c.get("/app/commit")
+        assert res.status_code == http.HTTPStatus.OK
+        assert "view_history" not in res.body.json()
+        txid = res.body.json()["transaction_id"]
+
+        # non-true view_history should still work
+        res = c.get("/app/commit?view_history=nottrue")
+        assert res.status_code == http.HTTPStatus.OK
+        assert "view_history" not in res.body.json()
+
+        # true view_history should list all history
+        res = c.get("/app/commit?view_history=true")
+        assert res.status_code == http.HTTPStatus.OK
+        assert "view_history" in res.body.json()
+        assert type(res.body.json()["view_history"]) == list
+        view_history = res.body.json()["view_history"]
+
+        # view_history should override the view_history_since
+        res = c.get("/app/commit?view_history=true&view_history_since=2")
+        assert res.status_code == http.HTTPStatus.OK
+        assert res.body.json() == {"transaction_id": txid, "view_history": view_history}
+
+        # ask for since the current view
+        res = c.get("/app/commit?view_history_since=2")
+        assert res.status_code == http.HTTPStatus.OK
+        view_history_2 = [v for v in view_history if int(v.split(".")[0]) > 2]
+        if view_history_2:
+            assert res.body.json() == {
+                "transaction_id": txid,
+                "view_history": view_history_2,
+            }
+        else:
+            assert "view_history" not in res.body.json()
+
+        # ask for an invalid view
+        res = c.get("/app/commit?view_history_since=0")
+        assert res.status_code == http.HTTPStatus.BAD_REQUEST
+        assert res.body.json() == {
+            "error": {
+                "code": "InvalidQueryParameterValue",
+                "message": "Error code: InvalidArgs",
+            }
+        }
+
+        # ask for something that doesn't exist yet
+        res = c.get("/app/commit?view_history_since=100")
```

Comment:
Arguably 404?

Reply:
Yep, changed future views to return 404.